webapp/utils/text_formatting.jsx

```diff
@@ -181,8 +192,7 @@ export function autolinkAtMentions(text, tokens, usernameMap) {
             const truncated = usernameLower.substring(0, c);
             const suffix = usernameLower.substring(c);
 
-            // If we've found a username or run out of punctuation to trim off, render it as an at mention
-            if (mentionExists(truncated) || !punctuation.test(truncated[truncated.length - 1])) {
+            if (mentionExists(truncated)) {
```

Comment:
Can you make it so that if `!punctuation.test(truncated[truncated.length - 1])`, then we break out of the loop? As it is now, if we have a user named `@will` and we message `@william`, this will get rendered as a link to `@will` since it chops off non-punctuation